airbyte-webapp/src/components/ResetDataModal/ResetDataModal.tsx

```diff
@@ -23,56 +24,61 @@ const ButtonContent = styled.div`
   padding-top: 27px;
   text-align: right;
 `;
-const ButtonWithMargin = styled(Button)`
+const ButtonWithMargin = styled(LoadingButton)`
   margin-right: 9px;
 `;
 
-const ResetDataModal: React.FC<IProps> = ({ onClose, onSubmit, modalType }) => {
-  const modalText = () => {
-    if (modalType === ModalTypes.RESET_CHANGED_COLUMN) {
-      return <FormattedMessage id="form.changedColumns" />;
-    }
-
-    if (modalType === ModalTypes.UPDATE_SCHEMA) {
-      return <FormattedMessage id="connection.updateSchemaText" />;
-    }
-
-    return <FormattedMessage id="form.resetDataText" />;
-  };
+interface IModalTextProps {
+  modalType: ModalTypes | undefined;
+}
+const ModalText: FC<IModalTextProps> = ({ modalType }) => {
+  if (modalType === ModalTypes.RESET_CHANGED_COLUMN) {
+    return <FormattedMessage id="form.changedColumns" />;
+  }
 
-  const modalTitle = () => {
-    if (modalType === ModalTypes.UPDATE_SCHEMA) {
-      return <FormattedMessage id="connection.updateSchema" />;
-    }
+  if (modalType === ModalTypes.UPDATE_SCHEMA) {
+    return <FormattedMessage id="connection.updateSchemaText" />;
+  }
 
-    return <FormattedMessage id="form.resetData" />;
-  };
+  return <FormattedMessage id="form.resetDataText" />;
+};
 
-  const modalCancelButtonText = () => {
-    if (modalType === ModalTypes.UPDATE_SCHEMA) {
-      return <FormattedMessage id="form.cancel" />;
-    }
+const ResetDataModal: React.FC<IProps> = ({ onClose, onSubmit, modalType }) => {
```

Comment:
We can probably replace this entire modal with the confirmationModalService instead.

Reply:
Oh, didn't notice that we have a new confirmationModalService. 
Yes, I think we indeed can replace it with this kind of service. 
Thanks for the hint 👍🏻